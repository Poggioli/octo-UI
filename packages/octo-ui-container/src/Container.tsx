import { forwardRef } from "@octo-ui/core";
import { StyledContainer } from "./styles";
import { ContainerProps } from "./types";

const Container = forwardRef<typeof StyledContainer, ContainerProps>(
  ({ ...props }, forwardedRef) => {
    return <StyledContainer {...props} ref={forwardedRef} />;
  }
);
Container.toString = () => `.${StyledContainer.className}`;

export { StyledContainer };
export { Container };

