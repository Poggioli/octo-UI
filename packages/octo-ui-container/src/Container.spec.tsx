import { render } from "@testing-library/react";
import { Container } from "./Container";

describe("<Container />", () => {
  it(`GIVEN a <Container />
      WHEN render is called
      THEN should render a <Container />`, () => {
    const rendered = render(<Container />);
    expect(rendered.container).toMatchSnapshot();
  });

  it(`GIVEN a <Container />
      WHEN toString is called
      THEN return a string`, () => {
    expect(Container.toString()).toBe(".octo-UI-c-fEDgXe");
  });
});
