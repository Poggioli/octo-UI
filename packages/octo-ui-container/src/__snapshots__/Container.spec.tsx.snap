// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container /> GIVEN a <Container />
      WHEN render is called
      THEN should render a <Container /> 1`] = `
<div>
  <div
    class="octo-UI-c-fEDgXe octo-UI-c-fEDgXe-gdopcj-size-4"
  />
</div>
`;
